wallet/src/main/java/ch/admin/bag/covidcertificate/wallet/detail/CertificateDetailItem.kt

```diff
@@ -26,42 +30,48 @@ data class TitleItem(@StringRes val titleResource: Int) : CertificateDetailItem(
 
 	override fun bindView(view: View) {
 		view.findViewById<TextView>(R.id.item_title).setText(titleResource)
+		val englishLabel = view.findViewById<TextView>(R.id.item_title_english)
+		englishLabel.isVisible = view.context.getString(R.string.language_key) != "en"
+		englishLabel.text = getEnglishTranslation(view.context, titleResource)
 	}
 }
 
-data class TitleStatusItem(@StringRes val labelResource: Int, val status: String) : CertificateDetailItem() {
-	companion object {
-		const val layoutResource = R.layout.item_detail_title_status
-	}
-
-	override fun bindView(view: View) {
-		view.findViewById<TextView>(R.id.item_title_status_label).setText(labelResource)
-		view.findViewById<TextView>(R.id.item_title_status_value).text = status
-	}
-}
-
-data class ValueItem(@StringRes val labelResource: Int, val value: String) : CertificateDetailItem() {
+data class ValueItem(@StringRes val labelResource: Int, val value: String, val isTranslatedToEnglish: Boolean = true) :
+	CertificateDetailItem() {
 	companion object {
 		const val layoutResource = R.layout.item_detail_value
 	}
 
 	override fun bindView(view: View) {
 		view.findViewById<TextView>(R.id.item_value_label).setText(labelResource)
 		view.findViewById<TextView>(R.id.item_value_value).text = value
+		val englishLabel = view.findViewById<TextView>(R.id.item_value_label_english)
+		englishLabel.isVisible = isTranslatedToEnglish && view.context.getString(R.string.language_key) != "en"
+		englishLabel.text = getEnglishTranslation(view.context, labelResource)
 	}
 }
 
-data class ValueItemWithoutLabel(val value: String) : CertificateDetailItem() {
+data class ValueItemWithoutLabel(val value: String, val isGrey: Boolean = false) : CertificateDetailItem() {
 	companion object {
 		const val layoutResource = R.layout.item_detail_value_without_label
 	}
 
 	override fun bindView(view: View) {
-		view.findViewById<TextView>(R.id.item_value_text).text = value
+		val valueText = view.findViewById<TextView>(R.id.item_value_text)
+		if (isGrey) {
+			valueText.setTextColor(view.context.getColor(R.color.grey))
```

Comment:
Since this is called from a RecyclerView ViewHolder which reuses its views, there should be an else case resetting the color to non-grey, otherwise it might happen that a recycled view which has `isGrey = false` still displays a grey text

Reply:
fixed